### PR TITLE
Add option `selectHiddenElement` - To Link Selector

### PR DESCRIPTION
**Pull request by jayandran-Sampath**
Add option `selectHiddenElement` to Link Selector, myself & @sathiyaraj worked on enabled Hidden Element Search for Link Selector, If we are good with this, i will do the same for remaining selectors

Part of #811 - link

### CODE REVIEW

File: lib/taiko.js [hunk 1]
```diff
@@ -2042,6 +2043,70 @@ module.exports.link = (attrValuePairs, ...args) => {
...
+  assertType(text);
+  let nbspChar = String.fromCharCode(160);
+  let textToTranslate = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' + nbspChar;
+  let translateTo = 'abcdefghijklmnopqrstuvwxyz' + ' ';
+  let xpathText = `translate(normalize-space(${xpath(
+      text,
+  )}), "${textToTranslate}", "${translateTo}")`;
+  const get = async () => await handleRelativeSearch(await $$xpath(
+      `//text()[contains(translate(normalize-space(string()), "${textToTranslate}", "${translateTo}"), ${xpathText})]`,
+  ),args);
```
Comment (saikrishna321) [inline lib/taiko.js]: Can't we use the exists fn - https://github.com/getgauge/taiko/pull/866/files#diff-fa054c0cd359737bbdc57062a6fee95aR122

Comment (saikrishna321) [reply lib/taiko.js]: Can the logic be moved to `elementSearch.js`? @nehashri Thoughts?

File: lib/taiko.js [hunk 2]
```diff
@@ -2042,6 +2043,70 @@ module.exports.link = (attrValuePairs, ...args) => {
...
+    assertType(text);
+    let nbspChar = String.fromCharCode(160);
+    let textToTranslate = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' + nbspChar;
+    let translateTo = 'abcdefghijklmnopqrstuvwxyz' + ' ';
+    let xpathText = `translate(normalize-space(${xpath(
+        text,
+    )}), "${textToTranslate}", "${translateTo}")`;
+    const get = async () => await handleRelativeSearch(await $$xpath(
+        `//text()[starts-with(translate(normalize-space(string()), "${textToTranslate}", "${translateTo}"), ${xpathText})]`,
+    ),args);
```
Comment (saikrishna321) [inline lib/taiko.js]: Same as above!

File: lib/taiko.js [hunk 3]
```diff
@@ -2042,6 +2043,70 @@ module.exports.link = (attrValuePairs, ...args) => {
...
+    return {
+        get: getIfExists(get),
+        exists: exists(getIfExists(get), descEvent),
+        description: `Element starts with text "${text}"`,
+        text: selectorText(get),
+    };
+};
```
Comment (negiDharmendra) [inline lib/taiko.js]: @jayandran-Sampath  Please separate these changes from this PR.

**Comment by jayandran-Sampath** [comment]
@negiDharmendra @saikrishna321 Sorry my bad, i raised this PR on top of existing PR, I will work on it properly and then raise it back. Thanks

File: lib/elementSearch.js [hunk 4]
```diff
@@ -95,46 +95,54 @@ async function matchContainsElements(
...
-  xpathText,
+  xpathText,options,
 ) {
-  return await $$xpath(
-    `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,
+  return options ? await $$xpath(
+    `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,options.selectHiddenElement
+  ) : await $$xpath(
+      `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,
   );
```
Comment (negiDharmendra) [inline lib/elementSearch.js]: We don't need this check here as [$$xpath](https://github.com/getgauge/taiko/blob/73243dee5cbddb9e69ae30b4391b8d26a288f20e/lib/elementSearch.js#L249) already takes care of this scenario. I.M.O we should just pass the `selectHiddenElement` option to the `$$xpath` function.

Comment (jayandran-Sampath) [reply lib/elementSearch.js]: @negiDharmendra thanks, i have tried without this condition but getting exception, thats why  handled it

**Pull request closed without merging**